### PR TITLE
Rebased: Add PreRenderDocument and PostRenderDocument events

### PR DESCRIPTION
**Pull request by linawolf**

### CODE REVIEW

File: packages/guides/src/Renderer/LatexRenderer.php [hunk 1]
```diff
@@ -28,7 +31,7 @@ public function supports(string $outputFormat): bool
 
     public function render(RenderCommand $renderCommand): void
     {
-        $renderDocumentHandler = new RenderDocumentHandler($this->renderer);
+        $renderDocumentHandler = new RenderDocumentHandler($this->renderer, $this->eventDispatcher);
```
Comment (jaapio) [inline packages/guides/src/Renderer/LatexRenderer.php:34]: This should also be removed as you did in the HtmlRenderer by using the commandbus.

**Review by jaapio — CHANGES_REQUESTED** [review]
Thanks for addressing some bugs in this PR! 

Just a small improvement that should be done before we can merge this.

File: packages/guides/src/Renderer/LatexRenderer.php (hunk 1 above)
Comment (linawolf) [reply packages/guides/src/Renderer/LatexRenderer.php:34]: @jaapio I don't really get yet what the command bus is doing and how to use it.

Comment (jaapio) [reply packages/guides/src/Renderer/LatexRenderer.php:34]: The commandbus is routing a command to a handler. 
Some background info:
https://barryvanveen.nl/articles/49-what-is-a-command-bus-and-why-should-you-use-it

**Comment by jaapio** [comment]
Yeah, yet another step forward! Thanks

**Pull request merged**